src/strings.cpp

```diff
@@ -2293,7 +2293,7 @@ void CheckForMissingGlyphs(bool base_font, MissingGlyphSearcher *searcher)
 			/* Our fallback font does miss characters too, so keep the
 			 * user chosen font as that is more likely to be any good than
 			 * the wild guess we made */
-			InitFontCache(searcher->Monospace());
+			InitFontCache();
```

Comment:
When searching for missing glyphs in the monospace font, this will now reinitialise all the font caches, not just monospace.

Reply:
That's correct. Is there a reason for the monospaced font to be special cased here?

Reply:
Yes, monospace font searching is only used when loading a text file. None of the language files make use of the monospace font.

Conversely, the normal font sizes aren't relevant when loading a text file.

This is also why the monospace font loading is normally deferred until a text file is loaded.

Reply:
> Conversely, the normal font sizes aren't relevant when loading a text file.

Some font (looks like the FS_NORMAL font) is used in the title bars of the display windows.  So this isn't true.

My understanding of your point is that the old code defers the monospaced font loading and for some cases that's better.  One less case in the future as the Monospaced font will need to be loaded for the font settings tab.

My counter argument is that the font code isn't in the critical path.  Optimization is nice but unneeded. The font code is already badly optimized with repeated calls to InitFontCache and CheckForMissingGlyphs.  Simplifying this block of code is worth the cost.